werkzeug/routing.py

```diff
@@ -765,7 +765,9 @@ def match(self, path):
                 # tells the map to redirect to the same url but with a
                 # trailing slash
                 if self.strict_slashes and not self.is_leaf and \
-                   not groups.pop('__suffix__'):
+                        not groups.pop('__suffix__') and \
+                        (method is None or self.methods is None or
```

Comment:
That's what I mean, but after thinking a bit more about it I now think it's better to be strict.